src/runtime/database.cpp

```diff
@@ -272,16 +272,77 @@ std::string Database::open(bool wait, bool memory, bool tty) {
     }
 #endif
 
-    char *fail;
+    // use PRAGMA user_version to store the schema version in the DB header for quick access.
+    int header_ver = 0;
+    {
+      sqlite3_stmt *st = nullptr;
+      if (sqlite3_prepare_v2(imp->db, "PRAGMA user_version;", -1, &st, nullptr) == SQLITE_OK &&
+          sqlite3_step(st) == SQLITE_ROW)
+        header_ver = sqlite3_column_int(st, 0);
+      sqlite3_finalize(st);
+    }
+
+    // since older wake.db files may not have the header set, we need to check the legacy table,
+    // to differentiate between a brand-new DB and an old one.
+    // Todo: remove this check once all wake.db instances have migrated to newer versions.
```

Comment:
I can't say I agree with this TODO since I don't believe we can guarantee at point in time when this will be true. 

One Nit: 
```suggestion
    // TODO: remove this check once all wake.db instances have migrated to newer versions.
```

Reply:
Removing the version table in the schema will help clean out the code, so we dont have to maintain it and check it in two places. I dont think we need all instances of the wake.db to have been migrated, just a decent amount of time to have been passed so that most users haver the user_version in their wake.db header. The only fallout of removing this check when there are still wake.db instances of the older version, is that they will still get a wake.db error related to the actual schema error- but i'll go ahead and add an error message that is less confusing for that error path as well.